### PR TITLE
Refactor: upload tests and modified get user function

### PR DESCRIPTION
**Pull request by annavaws**
## Description
Relocated test_upload.py, test_audio.mp3, and test_image.jpg into tests module. Additionally, the get_current_user function has been updated to obtain the access token from headers passed from the frontend instead of cookies.

### CODE REVIEW

**Comment by daffafaizan** [comment]
In my opinion, **upload** can be a part of the **transcription** tag so you can add your endpoints in the ```transcription.py``` controller rather than making a new ```upload.py``` file. It could be something like ```v1/transcription/upload```.

File: src/services/users.py [hunk 1]
```diff
@@ -27,12 +27,17 @@ def update_tokens(session: Session, user, access_token: str, refresh_token: str)
 
 
 def get_current_user(request: Request, session: Session = Depends(get_db)):
-    access_token = request.cookies.get("access_token")
+    access_token = request.headers.get("Authorization")
```
[the remark this answers is not in the document]
Comment (annavaws) [reply src/services/users.py:30]: Yes, when I refreshed the page, the cookies vanished. But, it remained on the localStorage, so I'm passing the token using a header.

File: tests/test_upload.py [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+    response = client.post("v1/auth/register", json=data)
+    assert response.status_code == 200
+    assert response.json()["email"] == "upload@example.com"
+
+
+def get_access_token(test_db):
+    """Helper function to get the access token by logging in"""
+    response_login = client.post(
+        "v1/auth/login", json={"email": "upload@example.com", "password": "StrongPassword123"})
```
[the remark this answers is not in the document]
Comment (annavaws) [reply tests/test_upload.py:22]: I agree, I'll change it

[the remark this answers is not in the document]
Comment (annavaws) [reply tests/test_upload.py]: Done

File: src/services/users.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (daffafaizan) [reply src/services/users.py:30]: Can you use an ```or``` operator? For example in Java,
```request.cookies.get("access_token") || request.headers.get("Authorization")```

**Comment by daffafaizan** [comment]
> In my opinion, **upload** can be a part of the **transcription** tag so you can add your endpoints in the `transcription.py` controller rather than making a new `upload.py` file. It could be something like `v1/transcription/upload`.

This will be tackled in a different time.

File: src/services/users.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (jeremyalv) [reply src/services/users.py:30]: Someone wrote a good explanation here, https://stackoverflow.com/questions/62999140/access-token-and-refresh-token-best-practices-how-to-implement-access-refres

They suggested to use Local storage to store access tokens, and pass it through a header. As long as we use **token rotation**, storing it in localstorage should be ok - https://stackoverflow.com/questions/57650692/where-to-store-the-refresh-token-on-the-client

**Comment by jeremyalv** [comment]
> > In my opinion, **upload** can be a part of the **transcription** tag so you can add your endpoints in the `transcription.py` controller rather than making a new `upload.py` file. It could be something like `v1/transcription/upload`.
> 
> This will be tackled in a different time.

I think `/upload` is more proper because we're not uploading transcriptions, but rather audio files to be transcribed later on.

**Review by jeremyalv — APPROVED** [review]
For the access token I've added my thoughts in the comments. Looks good overall.

File: src/services/users.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (annavaws) [reply src/services/users.py:30]: Yeah, I also think it's better to pass through a header.

**Pull request merged**